### DISCUSSION TITLE
OnClick Event in React Calendar

### DISCUSSION THREAD

**Discussion by therealchrismcfly** [neuefische/questions#113] [Web] [answered]
Hallihallo,

ich habe meinen Kalender mit react-Calender eingebaut. Nun möchte ich, dass bei Klick auf Datum eine Funktion ausgelöst wird. Hab schon viel probiert und gegoogelt, aber hänge total fest. Kann jemand bitte in Raum 3 kommen und mir weiterhelfen?

Hier der Code zu meinem Repo:
https://github.com/therealchrismcfly/capstone-project/tree/calendar

Hier der Screenshot der Calender Komponente, um die es geht: 
![Bildschirmfoto 2022-08-26 um 10 57 59](https://user-images.githubusercontent.com/107792731/186867651-9681529b-8baf-4b81-b68c-f7b31427337c.png)

Danke schonmal! 

xoxo
Chris

**Comment by ahohnsen** [comment, marked as answer]
Hey Chris,

Instead of calling `setDate` on the `onChange` event, you can call another function. This function could for example set the date and execute further commands:
```jsx

	function handleChange(selectedDate) {
		setDate(selectedDate);

	}

	return (
		<>
			<h1>React Calendar</h1>
			<Calendar onChange={handleChange} value={date} />
			<p>SelectedDate: {date.toDateString()}</p>
		</>
	);
```
